src/App.tsx

```diff
@@ -1,27 +1,63 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
 
-// import { getAll, get5First, getRed } from './api/goods';
+import { getAll, get5First, getRedGoods } from './api/goods';
+import { Good } from './types/Good';
 // or
 // import * as goodsAPI from './api/goods';
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
 
-    <button type="button" data-cy="all-button">
-      Load all goods
-    </button>
+  const handleGetAllSubmit = () => {
+    getAll().then(goodsFromServ => {
+      setGoods(goodsFromServ);
+    });
+  };
 
-    <button type="button" data-cy="first-five-button">
-      Load 5 first goods
-    </button>
+  const handleGetFiveFirstSubmit = () => {
+    get5First().then(goodsFromServ => {
+      setGoods(goodsFromServ);
+    });
+  };
 
-    <button type="button" data-cy="red-button">
-      Load red goods
-    </button>
+  const handleGetRedSubmit = () => {
+    getRedGoods().then(goodsFromServ => {
+      setGoods(goodsFromServ);
+    });
+  };
```

Comment:
will be better to create one general function that will accept needed method for making server request and then store the result in state for avoiding code repeating